lib/pluginmanager/sources/base.rb

```diff
@@ -0,0 +1,47 @@
+# encoding: utf-8
+require "rubygems/package"
+
+module LogStash::PluginManager::Sources
+
+  class Base
+
+    attr_reader :uri
+
+    def initialize(uri)
+      @uri = URI(uri)
+    end
+
+    def exist?
+      raise "NotImplemented"
+    end
+
+    def fetch(dest="")
+      raise "NotImplemented"
+    end
+
+    def version
+      File.basename("#{uri}", ".zip").split("-")[1]
+    end
+
+    def to_s
+      "#{uri}"
+    end
+
+    def valid?
+      return false if version.nil? || version.empty?
+      is_zip_file? && valid_version?
+    end
+
+    def valid_version?
+      LOGSTASH_VERSION == version
+    end
```

Comment:
I find it a bit fragile to check for compatibility by using a version in the name of the package, is all the other xpack implementation doing the same thing?